### ISSUE TITLE
...nvim-tree/view.lua:345 Vim:E444: Cannot close last window

### ISSUE THREAD

**Issue by MrAustin2u** [kyazdani42/nvim-tree.lua#745]
1) Navigate to my project directory. Nvim-tree displays instead of Netrw(intended behavior)
2) Open telescope git_files
3) Select file
4) Attempt to open nvim-tree `map('n', 'nf', ':NvimTreeFindFile<CR>', { noremap = false, silent =  true })`
5) Receive this error in the command line `...nvim-tree/view.lua:345 Vim:E44: Cannot close last window`

However, if I navigate to a file from the root. I am able to toggle nvim-tree fine.

Lua config: 

```
 local function setup_nvim_tree()
   vim.g.nvim_tree_icons = {
     default = "",
     git = {
       unstaged = "",
       staged = "",
       unmerged = "",
       renamed = "",
       untracked = "",
       deleted = "",
     },
   }
   vim.g.nvim_tree_special_files = {}
	vim.g.nvim_tree_indent_markers = 1
	vim.g.nvim_tree_gitignore = 0
	vim.g.nvim_tree_highlight_opened_files = 2
	vim.g.nvim_tree_ignore = { ".git", "node_modules", ".cache", ".vscode", ".DS_Store", ".un" }
	
local present, nvimtree = pcall(require, "nvim-tree")
	if not present then
		return
	end

	nvimtree.setup({
    ignore_ft_on_setup = { "dashboard", "terminal", "quickfix" },
		view = {
			allow_resize = true,
			width = 45,
      side = "right",
      hide_root_folder = false
		},
    auto_close = true,
    disable_netrw = true,
		hijack_cursor = true,
    update_cwd = true,
	})
end
```

Lua mappings: 
```
map('n', '<C-n>', ':NvimTreeToggle<CR>', { noremap = true })
map('n', 'nf', ':NvimTreeFindFile<CR>', { noremap = false, silent =  true })
map('n', 'nc', ':NvimTreeClose<CR>', { noremap = false, silent =  true })
map('n', 'nr', ':NvimTreeRefresh<CR>', { noremap = false, silent = true })
```

**Comment by kyazdani42** [comment]
@nyngwang please do not make such useless and aggressive comments, its really not nice...
@MrAustin2u i'll fix this.

**Comment by kyazdani42** [comment]
I cannot manage to reproduce with your config, can you produce a minimal configuration (with less to no plugins) where you can reproduce this consistently ?

**Comment by nyngwang** [comment]
> @nyngwang please do not make such useless and aggressive comments, its really not nice... @MrAustin2u i'll fix this.

Sorry, just updated my comment. But recently I spent all my energy debugging a similar issue. My point is simple: People will try to install many fancy plugins(like I did), but those plugins might cause trouble on the other plugins that really improve productivity like this one.

**Comment by MrAustin2u** [comment]
@kyazdani42 Minimal config is working just fine. Sorry for the false alarm. But thank you for looking into this.

**Comment by MrAustin2u** [comment]
> > @nyngwang please do not make such useless and aggressive comments, its really not nice... @MrAustin2u i'll fix this.
> 
> Sorry, just updated my comment. But recently I spent all my energy debugging a similar issue. My point is simple: People will try to install fancy plugins(like I did), but those plugins might cause trouble on the other plugins that really improve productivity(like this one).
> 
> For example, I did help another poor guy who also spent 5 hours on a bug similar to the OP: [sunjon/Shade.nvim#23](https://github.com/sunjon/Shade.nvim/issues/23) This is an example of "fancy plugin" I mentioned above. We're the users, if some plugins are not well written I think we have the right to blame the plugin itself (don't have to take it personally).

I also, got rid of Shade.nvim. Thank you for pointing this out @nyngwang

**Comment by kyazdani42** [comment]
@nyngwang ehm sorry i misunderstood :smile: i know plugins are not always perfect, but sometimes its better to try report a proper issue so the author can fix it instead of trashing the plugin :)

**Comment by kyazdani42** [comment]
i'll try to see why shade causes this. It should not happen.

**Comment by nyngwang** [comment]
> @nyngwang ehm sorry i misunderstood 😄 i know plugins are not always perfect, but sometimes its better to try report a proper issue so the author can fix it instead of trashing the plugin :)

No need to be sorry, your plugin is great!

To describe the bug in plain English:

- When you open many buffers and want to close one --> it said that you cannot close the last buffer. (for close I mean `:bd`)
- When you have only one buffer --> you can close it as expected.

**Comment by kyazdani42** [comment]
mhhh indeed i've seen this issue, i fixed it in the git rewrite #743 which is not yet properly finished but will come soon.
